app.py

```diff
@@ -0,0 +1,1792 @@
+    data['positions_gained'] = data['positions_gained'] * 0.9
+    return data
+
+
+# Sækja tölfræði fyrir Hamilton og Verstappen
+hamilton_original = get_driver_stats("lewis-hamilton")
+verstappen_original = get_driver_stats("max-verstappen")
+
+# Stilla gögn fyrir grafið
+hamilton_values_scaled = scale_to_log(hamilton_original)
+verstappen_values_scaled = scale_to_log(verstappen_original)
+
+categories = ['Total Points', 'Fastest Laps', 'Positions Gained', 'Driver of the Day Awards', 'Grand Slams']
+categoriesSpider = ['Total Points', 'Fastest Laps', 'Positions Gained', 'Driver of the Day Awards', 'Grand Slams']
+
+# Finna vísitölur fyrir flokkana í categoriesSpider
+indices = [categories.index(cat) for cat in categoriesSpider]
+
+
+# Búa til kóngulógraf með stilltum gildum og upprunalegum hover texta
+def create_spider_chart():
+    categories = categoriesSpider
+
+    fig = go.Figure()
+
+    # Gögn fyrir Hamilton með sérsniðnum hover texta sem sýnir óstilluð gildi
+    fig.add_trace(go.Scatterpolar(
+        r=hamilton_values_scaled,
+        theta=categories,
+        fill='toself',
+        name='Lewis Hamilton',
+        hoverinfo='text',
+        hovertext=[f'{cat}: {val}' for cat, val in zip(categories, hamilton_original)]
+    ))
+
+    # Gögn fyrir Verstappen með sérsniðnum hover texta sem sýnir óstilluð gildi
+    fig.add_trace(go.Scatterpolar(
+        r=verstappen_values_scaled,
+        theta=categories,
+        fill='toself',
+        name='Max Verstappen',
+        hoverinfo='text',
+        hovertext=[f'{cat}: {val}' for cat, val in zip(categories, verstappen_original)]
+    ))
+
+    fig.update_layout(
+        polar=dict(
+            bgcolor="#f9f9f9",  # Ljós bakgrunnur í polar plot
+            angularaxis=dict(
+                tickfont=dict(size=14, color="black"),  # Stærri og dekkri texti fyrir flokka
+                linewidth=2,  # Dekkri línur fyrir flokka
+                linecolor="black",  # Svartar línur frá miðju í flokka
+                ticks='outside',  # Ticks utan á ásnum
+                ticklen=5  # Lengd ticks
+            ),
+            radialaxis=dict(
+                showticklabels=False,  # Fjarlægir tölur á radial ásnum
+                gridcolor="lightgray",  # Minna áberandi netlínur
+                gridwidth=0.8,  # Þynnri netlínur
+                linecolor="gray",  # Minna áberandi radial ás
+                linewidth=0.8  # Þynnri radial línur
+            )
+        ),
+        showlegend=True,
+        legend=dict(
+            font=dict(size=12),
+            orientation="h",
+            yanchor="bottom",
+            y=1.1,
+            xanchor="center",
+            x=0.5
+        ),
+        template="plotly_white"
+    )
+
+    return fig
+
+
+def get_points_progression(driver_id):
+    conn = sqlite3.connect(DB_PATH)
+    query = f"""
+    SELECT race_id, race_points
+    FROM hamilton_verstappen_all_time_data
+    WHERE driver_id = '{driver_id}'
+    ORDER BY race_id
```

Comment:
Reiknið cumsum og röðunina bara beint í SQL 

```suggestion
SELECT 
    race_id, 
    race_points, 
    SUM(race_points) OVER (ORDER BY race_id) AS cumulative_points,
    ROW_NUMBER() OVER (ORDER BY race_id) AS career_race
FROM hamilton_verstappen_all_time_data
WHERE driver_id = '{driver_id}'
ORDER BY race_id;
```